### ISSUE TITLE
NVDA unable to identify link in HTML messages in Outlook for Office 2019

### ISSUE THREAD

**Issue by chojiro** [nvaccess/nvda#11173]
NVDA does not consistently identify links in HTML-messages in Outlook 2019. NVDA reads the text, but not whether it is a link or not. Using NVDA+F7 to bring up the elements list shows only the links NVDA will properly identify, but not other links that are present. Pressing Shift+F10 to bring up the context menu for actual links shows that it is a link, but NVDA doesn't see it as such.

### Steps to reproduce:
1. Find an HTML message
2. Read the message and try finding a link inside of it, both using Insert+F7 and K for quick navigation.
### Actual behavior:
NVDA does not consistently in all HTML messages recognize something is a link when it is.
### Expected behavior:
NVDA should consistently recognize whether something is a link or not in HTML messages.
#### NVDA installed/portable/running from source:
Installed
#### NVDA version:
2020.1
#### Windows version:
Windows 10Ver2004 (64-bit) build 19041.264, also attempted with windows version 1909.
#### Name and version of other software in use when reproducing the issue:
Outlook 2019 for Office 2019 Standard.
#### Does the issue still occur after restarting your computer?
Yes
#### If addons are disabled, is your problem still occuring?
Yes
#### Did you try to run the COM registry fixing tool in NVDA menu / tools?
Yes

**Comment by Adriani90** [comment]
@chojiro do you still have this issue with NVDA 2023.2? I cannot reproduce in Outlook for Office 2016 and Office 365.
Please make sure you set the UIA for Microsoft Word in NVDA advanced settings to "only when necessary". This setting affects also Outlook.

**Comment by chojiro** [comment]
I generally always use UIA at this point.

In 2023.1 I had this issue, so when I was able to use Windows Mail consistently again I switched. It will take a bit to test, as I'll have to wait for the necessary emails to come through.

If it happens, I can provide a .msg file as it is not confidential if necessary. If I do not encounter the problem, I can close this if you like.

**Comment by chojiro** [comment]
Here is a .msg file where this occurs.

It happens at the top, where links with text like "konferencer" exists. Some links are identified and some are not, but identified as links if you use K to navigate or arrow left and right to the end or beginning of a link.
[🧩 Nu i København Den Offentlige Formidler- og Underviseruddannelse.zip](https://github.com/nvaccess/nvda/files/13713867/Nu.i.Kobenhavn.Den.Offentlige.Formidler-.og.Underviseruddannelse.zip)

**Comment by Adriani90** [comment]
@chojiro again my question: did you try with UIA set to "only when necessary" in advanced settings pannel of NVDA? Does this make a difference?

**Comment by chojiro** [comment]
@Adriani90 No, this doesn't make a difference.

**Comment by Adriani90** [comment]
Ok this is still reproducible in Office 365. It seems NVDA works as expected when using ctrl+up and down arrow, but not when using up and down arrow only. When using ctrl+up and down, NVDA reports the beginning and the end of each link, and it navigates to every cell when the links are rendered in a table. This works both in browse and focus mode. So you are on the safe side if you use ctrl+up and down arrow to navigate.

However, when using up and down arrow only:
1. in case links are in table cells, In focus mode NVDA skips table cells because it navigates row by row and so it reads only the content of the focused cell instead of  reading the whole row. This would be fixed if NVDA provided an option to read entire table rows when pressing up and down arrow.
2. In browse mode, NVDA navigates cell by cell but announces only the first link as "link", while the next links are announced as if they were plain text. This occurs if there is no empty cell between links.
3. In case links are not in table cells, they are also announced as plain text if there is no blank line between them, this occurs both in browse and focus mode.

Parts 2 and 3 describe the same sympthhoms like in #14956.


cc: @CyrilleB79, @michaelDCurran, @gerald-hartig for triaging.

**Comment by CyrilleB79** [comment]
@chojiro you write:
Thanks for having provided an example of message in https://github.com/nvaccess/nvda/issues/11173#issuecomment-1862429003.

You write in the initial description:
> Using NVDA+F7 to bring up the elements list shows only the links NVDA will properly identify, but not other links that are present.

If we take the message you've provided as an example, could you illustrate with a specific link which is present in the message but not in `NVDA+F7` dialog? Please also indicate if it's with UIA (UIA always) or legacy (UIA Only when necessary).

@Adriani90:
Yes, I think that we can keep #14956 to track point 2 and 3; the reason is probably the same.

Regarding point 1, i.e. only one cell of the row is read when the caret enters a table cell with up/down arrow, I think we have already discussed this somewhere, but I cannot find any open or closed issue related to this. Are you able to mention it here (if it exists)?

**Comment by Mister-Kayne** [comment]
Today is my first day on Git Hub, and I am coming across so many issues related to NVDA here. I am also facing the similar issue but in my case the links list brings nothing up and the message is open but NVDA has stopped speaking. When reading the message also tabbing in Browse mode NVDA does not navigate to the links in the message. The UIA setting under advanced that was recommended in the thread of comments has also been tried but did not help. I am still going to trouble shoot and if I am successful, you will hopefully hear from me here. I managed to get a screen recording attaching it for your reference
